scripts/refresh-assets.ts

```diff
@@ -480,6 +499,100 @@ async function copyVersionToDotCom() {
 	)
 }
 
+class Code {
+	protected parts: (string | { toString(): string })[] = []
+
+	toString() {
+		return this.parts.map((part) => (typeof part === 'string' ? part : part.toString())).join('')
+	}
+
+	append(...parts: string[]): this {
+		for (const part of parts) {
+			this.parts.push(part + '\n')
+		}
+		return this
+	}
+}
+class CodeFile extends Code {
+	private nameId = 0
+	private imports = new Map<string, string>()
+
+	constructor(private header: string) {
+		super()
+	}
+
+	override toString(): string {
+		return [
+			this.header,
+			Array.from(this.imports, ([file, name]) => {
+				return `import ${name} from ${JSON.stringify(file)};`
+			}).join('\n'),
+			super.toString(),
+		].join('\n')
+	}
+
+	import(file: string) {
+		let name = this.imports.get(file)
+		if (!name) {
+			name = this.getName(file)
+			this.imports.set(file, name)
+		}
+		return name
+	}
+
+	getName(name: string) {
+		return `$_${name.replace(/[^a-zA-Z0-9_]+/g, '_')}_${this.nameId++}`
```

Comment:
any reason not to just use `/\W/g`?

Reply:
i can never remember what these single-letter shortcuts correspond to and have to look them up every time 😅 changed